openresty-patches/patches/1.15.8.3/LuaJIT-2.1-20190507_03-dyn-lightuserdata-map.patch

```diff
@@ -74,36 +75,35 @@ index 13b8230..05eed3a 100644
      void *p = lj_ccallback_new(cts, d, funcV(o));
      if (p) {
 diff --git a/LuaJIT-2.1-20190507/src/lj_obj.h b/LuaJIT-2.1-20190507/src/lj_obj.h
-index a89ea0d..77a157b 100644
+index a89ea0d9..92b62799 100644
 --- a/LuaJIT-2.1-20190507/src/lj_obj.h
 +++ b/LuaJIT-2.1-20190507/src/lj_obj.h
-@@ -830,6 +830,21 @@ typedef union GCobj {
+@@ -830,6 +830,20 @@ typedef union GCobj {
  #define setpriV(o, i)		(setitype((o), (i)))
  #endif
  
 +#ifdef LJ_LUV_SECT_SHIFT
-+#define LJ_LUV_SECT_MASK (7UL << LJ_LUV_SECT_SHIFT)
-+extern uint8_t luv_sect_map[8];
++#define LJ_LUV_SECT_MASK (((1UL << LJ_LUV_SECT_BITS) -1UL) << LJ_LUV_SECT_SHIFT)
++extern uint8_t luv_sect_map[1 << LJ_LUV_SECT_BITS];
```

Comment:
If we declare `LJ_LUV_NUM_SECTS` in this file, we can use it in these calculations instead of `1 << LJ_LUV_SECT_BITS`, which would make them a bit more clear IMHO. I don't know how to do that with `UL`.

Reply:
done